controllers/templatesync/template_sync.go

```diff
@@ -605,6 +610,48 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 	return reconcile.Result{}, resultError
 }
 
+// equivalentTemplates determines whether the template existing on the cluster and the policy template are the same
+func equivalentTemplates(eObject *unstructured.Unstructured, tObject *unstructured.Unstructured) bool {
+	if !equality.Semantic.DeepEqual(eObject.UnstructuredContent()["spec"], tObject.Object["spec"]) {
+		return false
+	}
+
+	if !equality.Semantic.DeepEqual(eObject.GetAnnotations(), tObject.GetAnnotations()) {
+		return false
+	}
+
+	if !equality.Semantic.DeepEqual(eObject.GetLabels(), tObject.GetLabels()) {
+		return false
+	}
+
+	if !equality.Semantic.DeepEqual(eObject.GetOwnerReferences(), tObject.GetOwnerReferences()) {
+		return false
+	}
+
+	return true
+}
+
+// setDefaultTemplateLabels ensures the template contains all necessary labels for processing
+func setDefaultTemplateLabels(instance *policiesv1.Policy, labels map[string]string) map[string]string {
+	if labels == nil {
+		labels = map[string]string{}
+	}
+
+	desiredLabels := map[string]string{
+		parentPolicyLabel:            instance.GetName(),
+		"cluster-name":               instance.GetLabels()[common.ClusterNameLabel],
+		common.ClusterNameLabel:      instance.GetLabels()[common.ClusterNameLabel],
+		"cluster-namespace":          instance.GetLabels()[common.ClusterNamespaceLabel],
```

Comment:
@mprahl Will this battle with the propagator in hosted mode?

Reply:
I updated it - is it just the cluster-namespace label or the one below it as well?

Reply:
I think it'd be both, but I'll let @mprahl answer since I don't really have a good handle on hosted mode yet...

But I'm also wondering whether there should be a specialized check for this label in the `EquivalentTemplates` function for hosted mode so that it's not updated every time?

Reply:
@mprahl could I get an update on this?